File: java/src/main/java/ps_traning/programmers/level_1/소수_찾기.java
package ps_traning.programmers.level_1;

public class 소수_찾기 {
    public int solution(int n) {
        int answer = 0;
        boolean[] checked = new boolean[n + 1];
        for (int num = 2; num <= n; num++) {
            if (checked[num]) continue;
            answer++;
            for (int m = num; m <= n; m += num) {
                checked[m] = true;
            }
        }
        return answer;
    }
}

File: python/with_java/programmers/level 1/소수 찾기.py
def solution(n):
    checked = [False for _ in range(n+1)]
    answer = 0
    for num in range(2, n+1):
        if checked[num]:
            continue
        answer += 1
        for multi in range(num, n+1, num):
            checked[multi] = True
    return answer